include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,81 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+This shared encoding is similar to SwizzledSharedEncodingAttr,
+but instead of repeating swizzling pattern every `blockSize = maxPhase*perPhase` rows of memory object,
+this layout changes swizzling pattern.
+
+Let's consider an element with position = (row, col).
+For simplicity, we do not vectorize memory,
```

Comment:
Is `vec=1` a fundamental limitation of the swizzling pattern or is it just for simplicity of an example? Good to be explicit here by saying "Use vec == 1 as an example for simplicity."

Reply:
It is just for simplicity of an example. vec groups multiple consecutive elements.
Original shared layout has a small example with vec=2, that show how it works.